### PR TITLE
add activation codes to existing users

### PR DESCRIPTION
**Pull request by tasawernawaz**
[LEARNER-2287](https://openedx.atlassian.net/browse/LEARNER-2287)

Please review @edx/learner-spartans

### CODE REVIEW

File: edx_sailthru/import_users_activation_codes/import_activation_codes.py [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+    parser.add_argument(
+        '--sailthru_secret',
+        help='Sailthru access secret.'
+    )
+
+    parser.add_argument(
+            '--mysql_host',
+            default=os.environ.get('EDXAPP_DB_HOST', ''),
+            help='Host address for ecommerce db replica.'
```
Comment (schenedx) [inline edx_sailthru/import_users_activation_codes/import_activation_codes.py]: Why do we want ecommerce db? The help text needs update?

File: edx_sailthru/import_users_activation_codes/import_activation_codes.py [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+        )
+
+    parser.add_argument(
+        '--mysql_user',
+        default=os.environ.get('EDXAPP_DB_USER', ''),
+        help='File where result is saved.'
```
Comment (schenedx) [inline edx_sailthru/import_users_activation_codes/import_activation_codes.py]: Help text needs update

File: edx_sailthru/import_users_activation_codes/import_activation_codes.py [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+    )
+
+    parser.add_argument(
+        '--mysql_pass',
+        default=os.environ.get('EDXAPP_DB_PASSWORD', ''),
+        help='Password for ecommerce db replica.'
```
Comment (schenedx) [inline edx_sailthru/import_users_activation_codes/import_activation_codes.py]: Again, help text needs update

File: edx_sailthru/import_users_activation_codes/import_activation_codes.py [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+        )
+
+    parser.add_argument(
+            '--db',
+            default=os.environ.get('EDXAPP_DB_NAME', ''),
+            help='File where course content is saved.'
```
Comment (schenedx) [inline edx_sailthru/import_users_activation_codes/import_activation_codes.py]: help text needs update

File: edx_sailthru/import_users_activation_codes/import_activation_codes.py [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+
+        else:
+            error = response.get_error()
+            logger.error("Error: " + error.get_message())
+            logger.error("Status Code: " + str(response.get_status_code()))
+            logger.error("Error Code: " + str(error.get_error_code()))
+
+
+def get_args(argv):
+    parser = argparse.ArgumentParser(description='Sailthru import activation codes of users script')
```
Comment (schenedx) [inline edx_sailthru/import_users_activation_codes/import_activation_codes.py]: `Import activation codes of inactive users from LMS to Sailthru`

File: edx_sailthru/import_users_activation_codes/import_activation_codes.py [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+            help='File where course content is saved.'
+    )
+
+    return parser.parse_args()
+
+
+def main(argv):
+    args = get_args(argv)
+
+    user_data = get_list_of_inactive_users(args)
```
Comment (schenedx) [inline edx_sailthru/import_users_activation_codes/import_activation_codes.py]: Can we be a bit more programmable on whether we should get active user or inactive users?
Say we can pass in an optional parameter `is_active` into this function.
If `is_active` is `True`, get all the active users
If `is_active` is `False`, get all the inactive users
If `is_active` is `None`, get all the users regardless
Then expose a script option that controls the `is_active` flag?
This suggestion is optional. I'll leave up to you to see if this is needed.

Comment (tasawernawaz) [reply edx_sailthru/import_users_activation_codes/import_activation_codes.py]: Updated according to your suggestion. Please review.

File: edx_sailthru/import_users_activation_codes/import_activation_codes.py [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+
+    logger.info("Total inactive users are: %s." %len(user_data))
```
Comment (schenedx) [inline edx_sailthru/import_users_activation_codes/import_activation_codes.py]: Update this log message please. Maybe just log out `Total users are <number>`

File: edx_sailthru/import_users_activation_codes/import_activation_codes.py [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+
+    sailthru_client = SailthruClient(args.sailthru_key, args.sailthru_secret)
+    update_user_on_sailthru(users_data=user_data, client=sailthru_client)
+
+    logger.info("All inactive users updated!")
```
Comment (schenedx) [inline edx_sailthru/import_users_activation_codes/import_activation_codes.py]: Update this log message to remove `inactive`

File: edx_sailthru/import_users_activation_codes/import_activation_codes.py [hunk 9]
```diff
@@ -0,0 +1,136 @@
...
+
+    users_data = pandas.read_sql_query(sql, edxapp_db)
+
+    return users_data
+
+
+def update_user_on_sailthru(users_data, client):
+    logger.info("Starting updating users on sailthru")
+
+    for index, user in users_data.iterrows():
```
Comment (schenedx) [inline edx_sailthru/import_users_activation_codes/import_activation_codes.py]: Have you tried to use their batch/job interface recommended in the ticket?
you will be making millions of API calls this way. That's quite inefficient.

Comment (tasawernawaz) [reply edx_sailthru/import_users_activation_codes/import_activation_codes.py]: I read the docs related to it and it is stated that `If you want to apply the same update to all of the users specified, you should pass the update parameter.` So I believe we are updating every user with his/her own `activation_code` so we can do a bulk update.

**Pull request merged**